### PR TITLE
fix(ui): Gracefully shrink and truncate long names

### PR DESCRIPTION
**Pull request by shazlehu**
### Proposed Change

Truncates long names with ellipsis and shrinks font for names longer than 15 characters. Adds a tool tip to display the full name for truncated names. Resolves observIQ/iris#1378

##### Checklist
- [ ] Changes are tested
- [ ] CI has passed

### CODE REVIEW

**Review by Mrod1598 — APPROVED** [review]
image 
image
LGTM

File: ui/src/components/Cards/OverviewDestinationCard.tsx [hunk 1]
```diff
@@ -93,38 +94,43 @@ const OverviewDestinationCardComponent: React.FC<ResourceDestinationCardProps> =
...
                     component="div"
-                    fontWeight={400}
-                    fontSize={14}
-                    variant="overline"
+                    fontWeight={600}
+                    gutterBottom
+                    fontSize={cardName.length > 15 ? 11 : 16}
                   >
-                    Paused
+                    {cardName.length > 20 && !cardName.includes(" ")
```
Comment (dsvanlani) [inline ui/src/components/Cards/OverviewDestinationCard.tsx:117]: Looks like we do this a couple places, can we factor this out into a function in `ui/src/utils` ?

**Review by dsvanlani — CHANGES_REQUESTED** [review]

File: ui/src/components/Cards/OverviewDestinationCard.tsx (hunk 1 above)
Comment (shazlehu) [reply ui/src/components/Cards/OverviewDestinationCard.tsx]: Surely

**Review by dsvanlani — APPROVED** [review]
Working great!

**Pull request merged**